WebDevStudios/wd_s
# Run [npm run theme] after successful webpack build so sprites and pot files are built.

This hooks into Webpack, and when a build is emitted, we run `npm run theme` so that pot files and sprites are re-generated into the build folder.

Closes https://github.com/WebDevStudios/wd_s/issues/583

### DESCRIPTION

![image](https://user-images.githubusercontent.com/1753298/109370810-243aed80-785f-11eb-83cf-4e2de236e848.png)

This solves https://github.com/WebDevStudios/wd_s/issues/583 by ensuring that the WebPack build runs the commands we use to built the `sprite.svg` and `languages/*.pot` files. 

The issue in https://github.com/WebDevStudios/wd_s/issues/583 is that when you run `npm run theme` `theme:icons` is ran which generates the `build/images/icons/sprite.svg` file, but then on `npm run start` Webpack's `CopyWebpackPlugin` plugin essentially deletes `build/images` and copies `src/images/**` over. Ran in this order, the `build/images/icons/sprite.svg` file is deleted in the process and never generated again.

This solution fixes this by having Webpack run `npm run theme` after `CopyWebpackPlugin` plugin runs to re-generate the `build/images/icons/sprite.svg` file. 

It also fixes another underlying (and related issue) where the POT files in `build/languages/*.pot` are built once when `npm run them` runs on `postinstall` but are never regenerated again. Additionally this fix fixes that too since pot files are generated to the `build/languages/` folder on `npm run theme`. 

### STEPS TO VERIFY

Running `node@12` clone the directory and run `npm i` then run `npm run start` and e.g. modify an SVG in in `src/images/icon` and notice that unlike in https://github.com/WebDevStudios/wd_s/issues/583 the `build/images/sprite.svg` file is no longer deleted on `npm run start`.

### DOCUMENTATION

## Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

I don't think so.
